main.cpp

```diff
@@ -19,13 +19,38 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto* CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+
+        if (CastExpr->getExprLoc().isMacroID())
+            return;
+
+        if (CastExpr->getCastKind() == CK_ToVoid)
+            return;
+
+        auto ReplaceRange = CharSourceRange::getCharRange(CastExpr->getLParenLoc(),
```

Comment:
What happens if the code is changed like this:  
```suggestion
        auto ReplaceRange = CharSourceRange::getTokenRange(CastExpr->getLParenLoc(),
```

What is the difference between `getCharRange()` and `getTokenRange()`?

Reply:
Разница заключается в следующем: 
Через `getCharRange()` мы выполняем работу с каждым символом, а через `getTokenRange()` мы выполняем работу с токенами.

Reply:
Please provide some sample code that shows the difference between the two methods.

Reply:
Основная разница в том, что при использовании `getTokenRange` мы сможем считать полностью имя типа данных в отличии от `getCharRange`, где работа идет с символами и мы сможем считать тип данных неполностью (т.к. счет будет идти до последнего символа).